process/interceptors/multiDataInterceptor.go

```diff
@@ -46,11 +47,20 @@ func NewMultiDataInterceptor(
 		factory:     factory,
 		processor:   processor,
 		throttler:   throttler,
+		isForMe:     NewDefaultDataVerifier(),
 	}
 
 	return multiDataIntercept, nil
 }
 
+// SetIsDataForCurrentShardVerifier sets a different implementation for the data verifier
+func (mdi *MultiDataInterceptor) SetIsDataForCurrentShardVerifier(verifier process.InterceptedDataVerifier) {
+	if check.IfNil(verifier) {
+		return
```

Comment:
I will not silently handle this. Maybe return the error?

Reply:
done.